.github/workflows/eslint.yml

```diff
@@ -0,0 +1,54 @@
+name: ESLint Checker
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
```

Comment:
we may want to support 2.x branch also

Reply:
Hi @Hailong-am 

Sure. Should I add this branch right now, or could it be a separate issue and a separate PR? Because once a branch is added, then ESLint errors in all files would need to be fixed, the test will always fail if this is not ensured.

Reply:
we may want to run lint check on changed/new files only.

Reply:
I am not sure we need this workflow, since it's already part of lint-stage which has to happen before commit. Unless someone disabled it in a PR, but reviewer should never let it happen

Reply:
Hi @bowenlan-amzn 

Should we just close this PR as a similar code exists?

Reply:
Yes, we can close this one